src/hardware/reelmagic/driver.cpp

```diff
@@ -0,0 +1,1138 @@
+  ReelMagic_MediaPlayer_Handle media_handle = reg_bl;
+  const uint16_t subfunc                      = reg_cx;
+  const uint16_t param1                       = reg_ax; //filename_ptr for command 0x1 & hardcoded to 1 for command 9
+  const uint16_t param2                       = reg_dx; //filename_seg for command 0x1
+  try {
+   //clear all regs by default on return...
+   reg_ax = 0; reg_bx = 0; reg_cx = 0; reg_dx = 0;
+
+   const uint32_t driver_call_rv =
+	   FMPDRV_driver_call(command, media_handle, subfunc, param1, param2);
+   reg_ax = (uint16_t)(driver_call_rv & 0xFFFF); // low
+   reg_dx = (uint16_t)(driver_call_rv >> 16);    // high
+   APILOG_DCFILT(command,
+		 subfunc,
+		 "driver_call(%02Xh,%02Xh,%Xh,%Xh,%Xh)=%Xh",
+		 (unsigned)command,
+		 (unsigned)media_handle,
+		 (unsigned)subfunc,
+		 (unsigned)param1,
+		 (unsigned)param2,
+		 (unsigned)driver_call_rv);
+  }
+  catch ([[maybe_unused]] std::exception& ex) {
+    LOG(LOG_REELMAGIC, LOG_WARN)("Zeroing out INT return registers due to exception in driver_call(%02Xh,%02Xh,%Xh,%Xh,%Xh)", (unsigned)command, (unsigned)media_handle, (unsigned)subfunc, (unsigned)param1, (unsigned)param2);
+    reg_ax = 0; reg_bx = 0; reg_cx = 0; reg_dx = 0;
+  }
+  return CBRET_NONE;
+}
+
+class FMPDRV final : public Program {
+  public:
+  FMPDRV()
+  {
+    AddMessages();
+    help_detail = {HELP_Filter::Common,
+		   HELP_Category::Dosbox,
+		   HELP_CmdType::Program,
+		   "FMPDRV"};
+  }
+  void Run()
+  {
+    if (HelpRequested()) {
+      MoreOutputStrings output(*this);
+      output.AddString(MSG_Get("PROGRAM_FMPDRV_HELP_LONG"));
+      output.Display();
+      return;
+    }
+    WriteOut(MSG_Get("PROGRAM_FMPDRV_TITLE"),
+	     REELMAGIC_DRIVER_VERSION_MAJOR,
+	     REELMAGIC_DRIVER_VERSION_MINOR);
+
+    if (cmd->FindExist("/u")) {
+      // unload driver
+      if (_installedInterruptNumber == 0) {
+	WriteOut(MSG_Get("PROGRAM_FMPDRV_UNLOAD_FAILED_NOT_LOADED"));
+	return;
+      }
+      if (!_unloadAllowed) {
+	WriteOut(MSG_Get("PROGRAM_FMPDRV_UNLOAD_FAILED_BLOCKED"));
+	return;
+      }
+      FMPDRV_UninstallINTHandler();
+      WriteOut(MSG_Get("PROGRAM_FMPDRV_UNLOADED"));
+    } else {
+      // load driver
+      if (_installedInterruptNumber != 0) {
+	WriteOut(MSG_Get("PROGRAM_FMPDRV_LOAD_FAILED_ALREADY_LOADED"),
+		 _installedInterruptNumber);
+	return;
+      }
+      if (!FMPDRV_InstallINTHandler()) {
+	WriteOut(MSG_Get("PROGRAM_FMPDRV_LOAD_FAILED_INT_CONFLICT"));
+	return;
+      }
+      WriteOut(MSG_Get("PROGRAM_FMPDRV_LOADED"), _installedInterruptNumber);
+    }
+  }
+
+  private:
+  void AddMessages()
+  {
+    MSG_Add("PROGRAM_FMPDRV_HELP",
+	    "Loads or unloads the built-in ReelMagic Full Motion Player Driver.\n");
+    MSG_Add("PROGRAM_FMPDRV_HELP_LONG",
+	    "Loads or unloads the built-in ReelMagic Full Motion Player Driver.\n"
+	    "\n"
+	    "Usage:\n"
+	    "  [color=cyan]FMPDRV[reset]    -- loads the driver.\n"
```

Comment:
We should keep help strings consistent. Therefore:

`[color=cyan]FMPDRV[reset]` -> `[color=green]fmpdrv[reset]`

Also, I think I haven't seem `--` anywhere; usage should rather look like:

`  [color=green]fmpdrv[reset] [/u]`

and the meaning of _/u_ should be explained in the _Where:_ section. See some other command for reference, for example `program_mount.cpp`.

Reply:
It looks deliberately wordy describing the `/u` flag in the `Where:` block:

```
     "Usage:\n"
     "  [color=green]fmpdrv[reset] [/u]\n"
     "\n"
     "Where:\n"
     "  The default action (no arguments) loads the driver.\n"
     "  /u unloads the driver.\n"
     "\n"
```

It looks like the `mount` command's help also described the boolean-only flags in the Usage block:

![2022-11-15_10-20](https://user-images.githubusercontent.com/1557255/201996215-0670d109-503a-41bb-ae7b-06d1ec909fd1.png)

I'll try something like that. It will be more straight forward to read (instead of needing two sections - usage and where to simply say "load and unload").

Reply:
To the point, I think:

![2022-11-15_10-30](https://user-images.githubusercontent.com/1557255/201998602-40031495-53fc-413a-bb16-f60c88742d3f.png)

Here's some user-testing (repeated unloads and loads, and how it responds):

![2022-11-15_10-31](https://user-images.githubusercontent.com/1557255/201998679-9d5d3fd8-5717-44d7-8d96-6753e2b7779c.png)

And here's the mixer in action when the card is present and then disabled at runtime:

![2022-11-15_10-32](https://user-images.githubusercontent.com/1557255/201998780-90f3ecd8-d93c-44c5-9d15-82a5aab82c53.png)

Reply:
Ok, let’s keep it this way.

Reply:
Thanks for this help, @FeralChild64.
The driver's simplicity also made it tricky to follow our standardized :slightly_smiling_face: